Dev

jnogle opened the pull request:
Added initial access rules (Kyle--need to test--will open issue), added api endpoint to register user and corresponding function triggered to add this new user to database, and api endpoint to delete a user and corresponding function triggered to remove this user from the database

jnogle committed:
add function to write newly registered user to the firestore db

fedorareis committed:
Merge pull request #9 from CruCentralCoast/master

Merge Kyle's rules commit with my dev branch

fedorareis committed:
updated the general rule to exclude user documents, I think

jnogle committed:
added error handling to new user registration function

jnogle committed:
fixed  error

jnogle committed:
Wrote function to receive new user data and validate key user fields

jnogle committed:
Changed api endpoints to use express routing, added user deletion endpoint and corresponding cloud function to remove that user from database

fedorareis commented on functions/index.js:
```diff
@@ -1,8 +1,215 @@
+        }
+        return res.status(400).json(json_res);
+    }
+	if(typeof password === 'undefined' || password.length === 0) {
```
Does the auth section not do its own password validation?

fedorareis commented on functions/index.js:
```diff
@@ -1,8 +1,215 @@
+        }
+            return res.status(400).json(json_res);
+    }
+    if(password.length < 8 || password.length > 72) {
```
This should be moved to be right next to the other Password validation.

fedorareis commented on functions/index.js:
```diff
@@ -1,8 +1,215 @@
+* Function to remove a user from the database                      *
+* Triggered by user deletion                                       *
+********************************************************************/
+exports.removeUser = functions.auth.user().onDelete(event => {
```
I'm a little confused as to the purpose of this function and how it differs from the delete function above.

fedorareis reviewed (changes requested):
Address my other comments and fix the indentation on the index file. The indentation seems to be all over the place.

jnogle committed:
added requested changes

jnogle commented:
- Fixed indentation
- Moved password validation statements next to one another
- To answer your first question, no, the auth sdk methods to create users do not do any of their own password validation
- To the second question, firebase stores users in a user pool separate from any database. Our users will obviously be created in a user pool for authentication purposes, but also written to the database so we can store additional information that the user pool does not account for (you can only store username, password, phone, photo there). One of the delete functions I've written (the REST endpoint) deletes the user from the user pool, and the second one removes that user from the database (this is automatically triggered by the user being removed from the pool). You'll notice the same pattern for creating users/adding them to the database.
